src/Grids/conformal_cubed_sphere_face_grid.jl

```diff
@@ -0,0 +1,229 @@
+using CubedSphere
+using JLD2
+using OffsetArrays
+using Rotations
+
+using Oceananigans
+using Oceananigans.Grids
+using Oceananigans.Grids: AbstractGrid, R_Earth
+
+import Base: show
+
+struct ConformalCubedSphereFaceGrid{FT, TX, TY, TZ, A, R} <: AbstractGrid{FT, TX, TY, TZ}
+        Nx :: Int
+        Ny :: Int
+        Nz :: Int
+        Hx :: Int
+        Hy :: Int
+        Hz :: Int
+      λᶜᶜᵃ :: A
+      λᶠᶜᵃ :: A
+      λᶜᶠᵃ :: A
+      λᶠᶠᵃ :: A
+      φᶜᶜᵃ :: A
+      φᶠᶜᵃ :: A
+      φᶜᶠᵃ :: A
+      φᶠᶠᵃ :: A
+      zᵃᵃᶜ :: R
+      zᵃᵃᶠ :: R
+     Δxᶜᶜᵃ :: A
+     Δxᶠᶜᵃ :: A
+     Δxᶜᶠᵃ :: A
+     Δxᶠᶠᵃ :: A
+     Δyᶜᶜᵃ :: A
+     Δyᶜᶠᵃ :: A
+     Δyᶠᶜᵃ :: A
+     Δyᶠᶠᵃ :: A
+     Δz    :: FT
+     Azᶜᶜᵃ :: A
+     Azᶠᶜᵃ :: A
+     Azᶜᶠᵃ :: A
+     Azᶠᶠᵃ :: A
+    radius :: FT
+end
+
+function ConformalCubedSphereFaceGrid(FT = Float64; size, z,
+                                      topology = (Bounded, Bounded, Bounded),
+                                             ξ = (-1, 1),
+                                             η = (-1, 1),
+                                        radius = R_Earth,
+                                          halo = (1, 1, 1),
+                                      rotation = nothing)
+    TX, TY, TZ = topology
+    Nξ, Nη, Nz = size
+    Hx, Hy, Hz = halo
+
+    ## Use a regular rectilinear grid for the face of the cube
+
+    ξη_grid = RegularRectilinearGrid(FT, topology=topology, size=(Nξ, Nη, Nz), x=ξ, y=η, z=z, halo=halo)
```

Comment:
We are using η for sea-surface height. Is it OK to use it here too?

Reply:
We could stick with Nx and Ny?

Reply:
Right yeah I'm using (ξ, η) as (x, y) on the face.

It's totally local to the constructor and we have other similar uses of η in e.g. https://github.com/CliMA/Oceananigans.jl/blob/master/src/Operators/field_metric_products.jl

But happy to change if we think it'll be better for the code!

Reply:
Blast. How about `x′, y′`?

Can also change the vorticity notation in `field_metric_products` to something else.